test/hotspot/jtreg/compiler/intrinsics/string/TestStringLatin1IndexOfChar.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * @test
+ * @bug 8173585
+ * @summary Test intrinsification of StringLatin1.indexOf(char). Note that
+ * differing code paths are taken contingent upon the length of the input String.
+ * Hence we must test against differing string lengths in order to validate
+ * correct functionality. We also ensure the strings are long enough to trigger
+ * the looping conditions of the individual code paths.
+ *
+ * Run with varing levels of AVX and SSE support, also without the intrinsic at all
+ *
+ * @library /compiler/patches /test/lib
+ * @run main/othervm -Xbatch -XX:Tier4InvocationThreshold=200 -XX:CompileThreshold=100 compiler.intrinsics.string.TestStringLatin1IndexOfChar
+ * @run main/othervm -Xbatch -XX:Tier4InvocationThreshold=200 -XX:CompileThreshold=100 -XX:+UnlockDiagnosticVMOptions -XX:DisableIntrinsic=_indexOfL_char compiler.intrinsics.string.TestStringLatin1IndexOfChar
+ * @run main/othervm -Xbatch -XX:Tier4InvocationThreshold=200 -XX:CompileThreshold=100 -XX:UseSSE=0 compiler.intrinsics.string.TestStringLatin1IndexOfChar
+ * @run main/othervm -Xbatch -XX:Tier4InvocationThreshold=200 -XX:CompileThreshold=100 -XX:UseAVX=1 compiler.intrinsics.string.TestStringLatin1IndexOfChar
+ * @run main/othervm -Xbatch -XX:Tier4InvocationThreshold=200 -XX:CompileThreshold=100 -XX:UseAVX=2 compiler.intrinsics.string.TestStringLatin1IndexOfChar
+ * @run main/othervm -Xbatch -XX:Tier4InvocationThreshold=200 -XX:CompileThreshold=100 -XX:UseAVX=3 compiler.intrinsics.string.TestStringLatin1IndexOfChar
+ */
+
+package compiler.intrinsics.string;
+
+import jdk.test.lib.Asserts;
+
+public class TestStringLatin1IndexOfChar{
```

Comment:
Can you please add testing for these edge cases:
- when the search char is the first char 
- when the search char is the last char 
- when the string has length 1

Reply:
Thanks for reviewing this. I have added a new test: `findOneItem` which covers these edge cases